source/server/options_impl_platform_linux.cc

```diff
@@ -0,0 +1,37 @@
+#include "server/options_impl_platform_linux.h"
+
+#include <sched.h>
+
+#include <thread>
+
+#include "server/options_impl_platform.h"
+
+namespace Envoy {
+
+uint32_t OptionsImplPlatformLinux::getCpuAffinityCount(unsigned int hw_threads) {
+  unsigned int threads = 0;
+  pid_t pid = getpid();
+  cpu_set_t mask;
+
+  CPU_ZERO(&mask);
+  if (sched_getaffinity(pid, sizeof(cpu_set_t), &mask) == -1) {
```

Comment:
Can this ever fail? Should we just RELEASE_ASSERT here? If it can fail, should we move this using OsSysCalls or some other or a LinuxOsSysCalls, etc.?

Reply:
It can fail, yes. I think the issue is systems with >1024 cpus, for which the extended `cpu_set_t` size needs to be iterated. I'll add the function to `LinuxOsSysCalls`. I have a patch for this now which will require some feedback, since I'm not sure if you meant the `OsSysCalls` interface to be extended by composition or just by regular inheritance.